### PR TITLE
ML&PL update post

### PR DESCRIPTION
**Pull request by MikeInnes**

### CODE REVIEW

**Comment by cormullion** [comment]
you have a "the the" on line 65.. :)

File: blog/_posts/2018-11-23-ml-compiler.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+All of these projects have enormous potential, but we think Julia has an edge. This post, based on our [paper to be presented at NIPS MLSys](https://arxiv.org/abs/1811.01457), will explore how we have used Julia to re-think ML tooling from the ground up, and provides some insight into the work that modern ML frameworks need to do.
+
+## Enter Flux
+
+We need a language to write differentiable algorithms, and Flux takes Julia to be this language. Being designed from the ground up for mathematical and numerical computing, Julia is unusually well-suited for expressing ML algorithms.  Meanwhile, its mix of modern design and new ideas in the compiler makes it easier to address the high performance needs of cutting edge ML.
+
+Where typical frameworks are all-encompassing monoliths in millions of lines of C++, Flux is only a thousand lines of straightforward Julia code that plays well with the rest of the ecosystem. Simply take [one package for taking gradients](https://github.com/FluxML/Zygote.jl), [one package for GPU support](https://github.com/JuliaGPU/CuArrays.jl/), sprinkle with some light convenience functions, bake for fifteen minutes and out pops a fully-featured ML stack.
```
Comment (jekbradbury) [inline blog/_posts/2018-11-23-ml-compiler.md]: TensorFlow is 1.5 million lines of C++; PyTorch is about 500,000 (including 220k in the Caffe2 tree). Still absolutely insane but not at the level of e.g. LLVM (8 million); and probably best to say "hundreds of thousands" rather than millions?

File: blog/_posts/2018-11-23-ml-compiler.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+A key advantage of Julia for this task is that it can be used to implement fundamental numerical libraries, like [differential equations solvers](http://juliadiffeq.org/) or [optimisation libraries](https://github.com/JuliaOpt/JuMP.jl); this neatly solves a growing need in the ML community, in which researchers backpropagate through high-performance code such as [ray tracers](https://people.csail.mit.edu/tzumao/diffrt/) and [physics engines](https://arxiv.org/abs/1611.01652), but the gradients must still be implemented by hand in C++. In contrast, since Julia’s implementations are written in Julia, everything from [ODEs](https://github.com/FluxML/model-zoo/blob/a243e8b192236c30064fcdb7a36f17f3b6823c34/other/diffeq/diffeq.jl) to [financial princing models](https://wilmott.com/automatic-for-the-greeks/) can be differentiated with ease.
```
Comment (jekbradbury) [inline blog/_posts/2018-11-23-ml-compiler.md]:
```suggestion
A key advantage of Julia for this task is that it can be used to implement fundamental numerical libraries, like [differential equations solvers](http://juliadiffeq.org/) or [optimisation libraries](https://github.com/JuliaOpt/JuMP.jl); this neatly solves a growing need in the ML community, in which researchers backpropagate through high-performance code such as [ray tracers](https://people.csail.mit.edu/tzumao/diffrt/) and [physics engines](https://arxiv.org/abs/1611.01652), but the gradients must still be implemented by hand in C++. In contrast, since Julia’s implementations are written in Julia, everything from [ODEs](https://github.com/FluxML/model-zoo/blob/a243e8b192236c30064fcdb7a36f17f3b6823c34/other/diffeq/diffeq.jl) to [financial pricing models](https://wilmott.com/automatic-for-the-greeks/) can be differentiated with ease.
```

Comment (MikeInnes) [reply blog/_posts/2018-11-23-ml-compiler.md]: Ha, well, I have to believe that "financial princing" is a thing.

File: blog/_posts/2018-11-23-ml-compiler.md (hunk 1 above)
Comment (MikeInnes) [reply blog/_posts/2018-11-23-ml-compiler.md]: Ah, the old prosody/accuracy tradeoff. I've updated the post.

Comment (jekbradbury) [reply blog/_posts/2018-11-23-ml-compiler.md]: Could do “zillions” :)

**Comment by ViralBShah** [comment]
@denizyuret - Would be good to have your eyes on this.

**Comment by ViralBShah** [comment]
I think the GPU and TPU sections be together.
